src/test/java/io/tiledb/spark/TileDBDataSourceReadTest.java

```diff
@@ -276,38 +276,45 @@ public void testQueryCondition() throws TileDBError {
     sparseHeterogeneousArrayCreate2A(dimensions);
     sparseHeterogeneousArrayWrite(data);
 
-    for (String order : new String[] {"row-major", "TILEDB_ROW_MAJOR"}) {
+    // Run with Query Condition on/off
+    Boolean[] tiledbFiltering = {true, false};
+    for (Boolean flag : tiledbFiltering) {
       Dataset<Row> dfRead =
           session()
               .read()
               .format("io.tiledb.spark")
-              .option("order", order)
-              .option("partition_count", 1)
+              .option("order", "row-major")
+              .option("partition_count", 2)
+              .option("tiledb_filtering", flag)
               .load(SPARSE_ARRAY_URI);
       dfRead.createOrReplaceTempView("tmp");
       dfRead.show();
       List<Row> rows1 = dfRead.sqlContext().sql("SELECT * FROM tmp").collectAsList();
       List<Row> rows2 =
           dfRead
               .sqlContext()
-              .sql("SELECT * FROM tmp WHERE a1 > 25")
+              .sql("SELECT * FROM tmp WHERE a2 >= 100 AND a1 > 25 ")
               .collectAsList(); // is pushed down
       List<Row> rows3 =
           dfRead
               .sqlContext()
-              .sql("SELECT * FROM tmp WHERE a1 >= 23")
+              .sql("SELECT * FROM tmp WHERE a1 >= 24 OR a1 = 23")
```

Comment:
Do we support compound expressions in the OR clause? Can we add a test for that?

Reply:
Can you give me an example?

Reply:
Like this, but with OR instead of AND: https://github.com/TileDB-Inc/TileDB-Py/blob/dev/tiledb/tests/test_query_condition.py#L285

Reply:
If not, no problem

Reply:
We do yes, I just updated the test to add this case.